compiler/daml-extension/src/extension.ts

```diff
@@ -320,13 +339,21 @@ class VirtualResourceManager {
         const panel = this._panels.get(uri);
         if (panel) {
             panel.webview.html = contents;
+            panel.webview.postMessage({command: 'del_note'});
```

Comment:
I don’t think you need to send this. Setting the `html` should reset web panel completely.

Reply:
This is what I started with, but notes were not cleared on `setContent`. It turned out to be caching mechanism, which was not reloading the page on consecutive `setContent` with the same input (as notes are added in JS) even though the DOM has changed in the meantime. Finally I removed this explicit notes clearing in favour of appending a HTML comment /w timestamp to the HTML content set in the `setContent`, so that this callback always performs a refresh.

Reply:
https://github.com/digital-asset/daml/pull/2141/files#diff-dcb66422b3cf60ef53cbc28d1bd8cd01R342

Reply:
Thanks for the explanation!